.github/workflows/reusable-pr-clone.yml

```diff
@@ -0,0 +1,144 @@
+      TARGET_BRANCH: ${{ inputs.head_ref }}
+      COMMIT_HASH:  ${{ inputs.sha }}
+      BASE_BRANCH: ${{ inputs.ref_name }}
+      SOURCE_PR: ${{ inputs.html_url }}
+      GIT_NAME: ${{ secrets.GIT_NAME }}
+      GIT_EMAIL: ${{ secrets.GIT_EMAIL}}
+    steps:
+      - name: Checkout repository CE
+        uses: actions/checkout@v4
+        with:
+            ref: ${{ inputs.head_ref }}
+            token: ${{ secrets.SYNC_TOKEN }}
+            fetch-depth: 0
+
+      - name: Set up Git
+        run: |
+          git config user.email "$GIT_EMAIL"
+          git config user.name "$GIT_NAME"
+
+      - name: Prepare base target branch PR in EE repo
+        run: |
+            SHORT_SHA=$(git rev-parse --short "$COMMIT_HASH")
+            echo "SHORT_SHA=$SHORT_SHA" >> $GITHUB_ENV
+            DESTINATION_BRANCH="$TARGET_BRANCH-from-ce-$SHORT_SHA"
+            echo "DESTINATION_BRANCH=$DESTINATION_BRANCH" >> $GITHUB_ENV
+
+            # Ensure we're working with the most recent code and branches
+            git fetch origin
+            git fetch https://github.com/pimcore/$TARGET_REPO.git
+
+            # Cleanup old remote if it exists and create a fresh remote reference
+            git remote | grep $TARGET_REPO && git remote remove $TARGET_REPO
+            git remote add $TARGET_REPO https://github.com/pimcore/$TARGET_REPO.git
+
+            # Check does base target branch exists in target repo and creates it if it doesnt exist
+            REPO_RESPONSE=$(curl -X GET --location "https://api.github.com/repos/pimcore/$TARGET_REPO/branches/$BASE_BRANCH" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer $PAT")
+            
+            BASE_TARGET_BRANCH=$(echo $REPO_RESPONSE | jq -r '.name')
+            if [ -z "$BASE_TARGET_BRANCH" ] || [ "$BASE_TARGET_BRANCH" = "null" ]; then
+              echo "BASE_TARGET_BRANCH $BASE_BRANCH is missing on target. Creating..."
+              git checkout $BASE_BRANCH
+              git pull
+            else
+              echo "$BASE_BRANCH exists on target. Skipping ..."
+            fi
+            
+            # Checkout and push to the destination branch in the target repo
+            git checkout $TARGET_BRANCH
+            git pull
+            git checkout -b $DESTINATION_BRANCH
+            git push $TARGET_REPO $DESTINATION_BRANCH
+            
+      - name: Create PR in target repo
+        run: |
+            # Create a PR in Repo EE
+            REQ_BODY=$(cat << EOF
+            {
+              "title": "Automated PR from CE repository $SHORT_SHA",
+              "body": "This PR is automatically created from CE repository pull request $SOURCE_PR",
+              "head": "$DESTINATION_BRANCH",
+              "base": "$BASE_BRANCH"
+            }
+            EOF
+            )
+
+            CREATE_PR=$(curl -X POST --location "https://api.github.com/repos/pimcore/$TARGET_REPO/pulls" \
```

Comment:
Would it make sense to use GitHub CLI instead for better readability? 
https://cli.github.com/manual/gh_pr_create
There were also have the possibility to add the PRs to projects ...?

Reply:
hey @brusch 

We decided to stick with curl and sh way instead of cli because not all methods we used for this new workflows were supported with gh cli. 

Also it gets a bit more complicated for handling outputs from gh cli, with status codes its much more cleaner. 

Let me know is it necessary to switch to gh cli instead of api calls now - I would like to stick with 1 approach, not to have 2 of them, so either cli or api calls.

Reply:
Thanks for your explanation, I'm fine with cURL as well 😊

Reply:
cool @brusch 

I will close this PR then and proceed further